pkg/sql/drop_user.go

```diff
@@ -121,36 +138,41 @@ func (n *dropUserNode) startExec(params runParams) error {
 			tree.Name(name).Format(&nameList, tree.FmtSimple)
 		}
 		return pgerror.NewErrorf(pgerror.CodeGroupingError,
-			"cannot drop user%s %s: grants still exist on %s",
-			util.Pluralize(int64(nameList.Len())), nameList.String(), usedBy.String(),
+			"cannot drop user%s or role%s %s: grants still exist on %s",
```

Comment:
can't we precisely say whether the intent was to drop a user or a role here, by checking the isRole state?

Reply:
We could look them up the same way. However, we may still end up with a mix of users and roles, making the error still gross. Since the `revoke` statement is the same for users and roles, I think this is ok.